src/write_gcm.c

```diff
@@ -3568,15 +3568,20 @@ static void *wg_process_queue(wg_context_t *ctx, wg_queue_t *queue,
     }
     if (wg_update_stats(stats) != 0) {
       wg_some_error_occured_g = 1;
-      WARNING("%s: wg_update_stats failed.", this_plugin_name);
+      ERROR("%s: wg_update_stats failed.", this_plugin_name);
       break;
```

Comment:
I actually missed the original motivation for this PR. Is the thread crashing under mysterious circumstances? What are those circumstances?

But the point of these breaks is to make the code readable: to stop processing as soon as something bad happens. I think it makes for a pretty good "readability" experience. The reader can read from top to bottom without having to keep a lot of state in his/her head about what's happening. Namely if we get to this line, things are still good.

If you indeed want to somehow restart processing when something bad happens (I'm dubious but I'll go along with it for the sake of argument), the way I personally would structure the code is as follows. I'd add a higher level method that calls this method. Then if this method returns an error, the higher level method can do what it needs to to restart the thread, keep it going, whatever.

I think this accomplishes better readability than a chain of if-elses, or whatever you're proposing.